router/shop.py

```diff
@@ -0,0 +1,117 @@
+from fastapi import APIRouter
+from fastapi import Depends
+from fastapi import status
+from fastapi.encoders import jsonable_encoder
+from fastapi.responses import JSONResponse
+
+from model.shop import ReturnShopModel, CreateShopForm, UpdateShopForm
+from model.general import SuccessModel, ErrorModel
+from utils.db_process import get_all_result, execute_query, dict_to_sql_command, dict_delete_none
+
+router = APIRouter()
+
+@router.get("/shop", tags=["shop"], responses={
+    status.HTTP_200_OK: {
+        "model": ReturnShopModel
+    },
+    status.HTTP_404_NOT_FOUND: {
+        "model": ErrorModel
+    }
+})
+async def get_shop(
+    shop_uuid: str = None,
+    account_uuid: str = None
+):
+    if shop_uuid:
+        sql = """
+            SELECT * FROM `Shop` WHERE shop_uuid = %s;
+        """
+        result = get_all_result(sql, (shop_uuid,))
+    elif account_uuid:
+        sql = """
+            SELECT * FROM `Shop` WHERE account_uuid = %s;
+        """
+        result = get_all_result(sql, (account_uuid,))
+    else:
+        sql = """
+            SELECT * FROM `Shop`;
+        """
+        result = get_all_result(sql)
+    if result:
+        return JSONResponse(
+            status_code=status.HTTP_200_OK,
+            content=jsonable_encoder({
+                "msg": "Success",
+                "data": result
+            })
+        )
+    return JSONResponse(
+        status_code=status.HTTP_404_NOT_FOUND,
+        content=jsonable_encoder({
+            "msg": "Fail"
+        })
+    )
+
+@router.post("/shop", tags=["shop"], responses={
+    status.HTTP_200_OK: {
+        "model": SuccessModel
+    },
+    status.HTTP_404_NOT_FOUND: {
+        "model": ErrorModel
+    }
+})
+async def create_shop(
+    shop_form: CreateShopForm = Depends(CreateShopForm.as_form)
+):
+    shop_form = shop_form.model_dump()
+    sql = """
+        INSERT INTO `Shop`
+        VALUES (UUID(), %s, %s, %s, %s, DEFAULT);
+    """
+    result = execute_query(sql, tuple(shop_form.values()))
```

Comment:
It is NOT banning users from creating multi shops, which violates our data model.

Reply:
i'll set db account_uuid to unique in Shop table

Reply:
```sql
alter table Shop
    add constraint Shop_pk
        unique (account_uuid);
```
something like this?

Reply:
Is this solved?